File: app/src/main/java/no/joharei/flixr/network/models/Photosets.java
package no.joharei.flixr.network.models;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

@SuppressWarnings("unused")
public class Photosets {

    @SerializedName("cancreate")
    private int canCreate;
    private int page;
    private int pages;
    @SerializedName("perpage")
    private int perPage;
    private int total;
    @SerializedName("photoset")
    private List<Photoset> photosets = new ArrayList<>();

    public List<Photoset> getPhotosets() {
        return photosets;
    }

}

File: app/src/main/java/no/joharei/flixr/network/models/Photosets.kt
package no.joharei.flixr.network.models

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class PhotosetsResponse(val photosets: Photosets)

@JsonClass(generateAdapter = true)
data class Photosets(
    val page: Int,
    val pages: Int,
    val perpage: Int,
    val total: Int,
    @Json(name = "photoset")
    val photosets: List<Photoset>
)